pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -133,16 +140,18 @@ func validateInterval(cfg *setting.UnifiedAlertingSettings, interval time.Durati
 }
 
 // validateForInterval validates ApiRuleNode.For and converts it to time.Duration. If the field is not specified returns 0 if GrafanaManagedAlert.UID is empty and -1 if it is not.
-func validateForInterval(ruleNode *apimodels.PostableExtendedRuleNode) (time.Duration, error) {
-	if ruleNode.ApiRuleNode == nil || ruleNode.ApiRuleNode.For == nil {
+func validateDurationPointerField(ruleNode *apimodels.PostableExtendedRuleNode, fieldName string) (time.Duration, error) {
```

Comment:
I think I am in favour of code duplication here to avoid reflection, what do you think?